### PR TITLE
Fix the code used to detect aarch64 capabilities when we don't have getauxval()

### LINKED ISSUE #19825: openssl-3.1.0-alpha1 aarch64 build failed: warning: implicit declaration of function ‘_armv8_eor3_probe’; did you mean ‘_armv8_sm3_probe’?
compile options:

```sh
./Configure -static --cross-compile-prefix="${CROSS_HOST}-" --prefix="${CROSS_PREFIX}" linux-aarch64 --openssldir=/etc/ssl
```

Error:

```
aarch64-linux-musl-gcc  -I. -Iinclude -Iproviders/common/include -Iproviders/implementations/include  -DBSAES_ASM -DECP_NISTZ256_ASM -DKECCAK1600_ASM -DMD5_ASM -DOPENSSL_BN_ASM_MONT -DOPENSSL_CPUID_OBJ -DOPENSSL_SM3_ASM -DPOLY1305_ASM -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DSM4_ASM -DSTATIC_LEGACY -DVPAES_ASM -DVPSM4_ASM -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSLDIR="\"/etc/ssl\"" -DENGINESDIR="\"/cross_root/aarch64-linux-musl/lib/engines-3\"" -DMODULESDIR="\"/cross_root/aarch64-linux-musl/lib/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF crypto/libcrypto-lib-armcap.d.tmp -MT crypto/libcrypto-lib-armcap.o -c -o crypto/libcrypto-lib-armcap.o crypto/armcap.c
crypto/armcap.c: In function ‘OPENSSL_cpuid_setup’:
crypto/armcap.c:358:13: warning: implicit declaration of function ‘_armv8_eor3_probe’; did you mean ‘_armv8_sm3_probe’? [-Wimplicit-function-declaration]
  358 |             _armv8_eor3_probe();
      |             ^~~~~~~~~~~~~~~~~
      |             _armv8_sm3_probe
crypto/armcap.c:404:1: error: expected declaration or statement at end of input
  404 | }
      | ^
make[1]: *** [Makefile:8195: crypto/libcrypto-lib-armcap.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/usr/src/openssl-3.1.0-alpha1'
make: *** [Makefile:2539: build_sw] Error 2
Error: Process completed with exit code 2.
```

Full output logs can be found in: https://github.com/abcfy2/aria2-static-build/actions/runs/3606952543/jobs/6078781257

### PR DESCRIPTION
**Pull request by tom-cosgrove-arm**
Fixes #19825.

### CODE REVIEW

File: crypto/armcap.c [hunk 1]
```diff
@@ -354,6 +355,7 @@ void OPENSSL_cpuid_setup(void)
         if (sigsetjmp(ill_jmp, 1) == 0) {
             _armv8_sm3_probe();
             OPENSSL_armcap_P |= ARMV8_SM3;
+        }
```
Comment (mattcaswell) [inline crypto/armcap.c:358]: !!! How did this ever work?

**Review by mattcaswell — APPROVED** [review]

**Comment by mattcaswell** [comment]
Is this master/3.1 only? Or does it apply to 3.0 too?

File: crypto/armcap.c (hunk 1 above)
Comment (tom-cosgrove-arm) [reply crypto/armcap.c:358]: Yeah, it was clearly never even tested :(

Comment (tom-cosgrove-arm) [reply crypto/armcap.c:358]: (which I did do before this PR)

**Comment by tom-cosgrove-arm** [comment]
Fortunately this isn't in 3.0

**Review by t8m — APPROVED** [review]

File: crypto/armcap.c (hunk 1 above)
Comment (t8m) [reply crypto/armcap.c:358]: > Yeah, it was clearly never even tested :(

Which is why we fortunately have alpha and beta pre-releases. As we cannot test on all environments that might be possibly supported by OpenSSL code base. :grin: 

The message to anyone who is using OpenSSL on platform that is not on the primary or secondary list - test the prereleases if you want the final release to support your platform.

**Comment by t8m** [comment]
Merged to master and 3.1 branches. Thank you for your contribution.

**Pull request closed without merging**